cpp/box/Box.h

```diff
@@ -228,14 +228,17 @@ class Box
     /*! \param vecs Vectors of fractional coordinates between 0 and 1 within
      *         parallelepipedal box
      *  \param Nvecs Number of vectors
-     *  \param out The array in which to place the wrapped vectors.
+     *  \param out_data The array in which to place the wrapped vectors.
      */
-    void makeAbsolute(const vec3<float>* vecs, unsigned int Nvecs, vec3<float>* out) const
+    void makeAbsolutePython(nb_array<float, nb::shape<-1, 3>> vecs, unsigned int Nvecs,
+                            nb_array<float, nb::shape<-1, 3>> out) const
```

Comment:
nanobind will silently convert arrays that are not the right type, shape, or ordering. This is fine for input arrays, but the output array must not be converted, or else we write output to temporary copy.

I see the Python side wraps output arrays with: `_convert_array(..., allow_copy=False)`. Could we instead use the [`noconvert`](https://nanobind.readthedocs.io/en/latest/ndarray.html#arrays-and-function-overloads) method that `nanobind` provides? One would hope that checking at the C++ level is faster - and that check must already be performed. It would also be less code for us to maintain in freud (no `_convert_array` method and no multiple class to `_convert_array` in every method).